lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -0,0 +1,276 @@
+
+    def create_user(self, userid, name, group, password, email):
+        """Creates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        # check that we have all fields
+        if None in [name, group, password]:
+            return dict(
+                changed=False,
+                failed=True,
+                msg="state is present but one of the following is missing: group,password or name")
+
+        # get group id
+        group_id = self.group_id(group)
+
+        # create new user
+        url = '{api_url}/users'.format(api_url=self.api_url)
+        resource = {'userid': userid, 'name': name, 'password': password, 'group': {'id': group_id}}
+        if email is not None:
+            resource['email'] = email
+
+        try:
+            result = self.manageiq.client.post(url, action='create', resource=resource)
+        except Exception as e:
+            self.manageiq.module.fail_json(msg="Failed to create user {userid}: {error}".format(userid=userid, error=e))
+
+        return dict(
+            changed=True,
+            msg="Successfully created the user {userid}: {user_details}".format(userid=userid, user_details=result['results']))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            miq=dict(required=True, type='dict'),
+            userid=dict(required=True, type='str'),
+            name=dict(),
+            password=dict(no_log=True),
+            group=dict(),
+            email=dict(),
+            state=dict(choices=['absent', 'present'], default='present')
+        ),
+    )
+
+    userid = module.params['userid']
+    name = module.params['name']
+    password = module.params['password']
+    group = module.params['group']
+    email = module.params['email']
+    state = module.params['state']
+
+    manageiq = ManageIQ(module)
+    manageiq_user = ManageIQUser(manageiq)
+
+    user = manageiq_user.user(userid)
+
+    # user should not exist
+    if state == "absent":
+        # if we do not have a user, nothing to do
+        if not user:
+            res_args = dict(
+                changed=False,
+                msg="User {userid}: does not exist in manageiq".format(userid=userid),
+            )
+
+        # if we have a user, delete it
+        else:
+            res_args = manageiq_user.delete_user(userid)
+
+    # user shoult exist
+    if state == "present":
+        # if we do not have a user, create it
+        if not user:
+            res_args = manageiq_user.create_user(userid, name, group, password, email)
+
+        # if we have a user, edit it
+        else:
+            res_args = manageiq_user.edit_user(user, name, group, password, email)
+
+    module.exit_json(**res_args)
+
+
+if __name__ == "__main__":
```

Comment:
You have the 'check_client' function to figure out if the requirements are present, but never actually use it, so the module fails on in import exception.

I recommend you place a call in the class init so you dont have to remember to call it in every module.